modules/afsocket/afinet-dest.c

```diff
@@ -43,10 +43,10 @@
 #endif
 
 #if SYSLOG_NG_ENABLE_SPOOF_SOURCE
-#pragma GCC diagnostic push "-Wundef"
+#pragma GCC diagnostic push
```

Comment:
i do not like these at all, we can add further compatibility headers as well for these if you wish, please add your opinions

Reply:
you can also add the header's name like `ignore_warnings_libnet` or something like that o avoid accidental poping the command line options, something else

Reply:
i do not really get this, sorry

push and pop a given warning is not supported by all compilers, it works in all or nothing mode for e.g. for given versions of clang

this type of push balanced with a pop will guarantee a proper, exact state restoration in any case, no matter where the options came from, the state will be intact, and exactly the same as before the push and after the pop, I hope at least